### DISCUSSION TITLE
How to load next/font/google in Storybook iframes

### DISCUSSION THREAD

**Discussion by kamal-choudhary** [storybookjs/storybook#23039] [Help]
Hi, I'm using Storybook on a NextJS + Tailwind project.

I have added NextJS fonts in the `src/pages/_app.js` file like this:

```
import '@styles/globals.css';
import { Inter } from 'next/font/google';

const inter = Inter({
  subsets: ['latin'],
  variable: '--font-inter',
});

export default function MyApp({ Component, pageProps }) {
  return (
    <main
      className={`${inter.variable} font-sans`}
    >
      <Component {...pageProps} />
    </main>
  );
}

```

The problem I'm facing is that font variable is not loading in the story. Let's say I am testing a `Button` component in Storybook, if I inspect the `button` element, I see the font variable greyed indicating it is **not defined**. (It makes sense as the font variable is set on the `main` element which is not present in Storybook as story gets rendered in an iFrame)

How do I fix this?

Any help will be much appreciated. Thanks.

**PS: I'm using the `@storybook/nextjs` package.**

---
storybook: "^7.0.20"
next: "13.3.0"

**Comment by JurajKavka** [comment]
Try to place that code to `.storybook/preview.ts` but there is issue too ...

In docs [here](https://www.npmjs.com/package/@storybook/nextjs#nextjs-font-optimization) is nothing more that google fonts just works. My case - I have two fonts. I'm trying to preload them in preview.ts but still only one font is loaded and second not. Font, that is loaded is explicitly named in `GlobalStyles` (I'm using styled-components). The second one I have in `theme` object and it is never preloaded.

**Comment by kamal-choudhary** [comment]
@JurajKavka thanks for your help - but I am loading it like this:

**.storybook/preview.js**

```
import '../src/styles/globals.css'; // replace with the name of your tailwind css file

import { inter, openSans } from './../src/pages/_app';

/** @type { import('@storybook/react').Preview } */
const preview = {
  parameters: {
    actions: { argTypesRegex: '^on[A-Z].*' },
    controls: {
      matchers: {
        color: /(background|color)$/i,
        date: /Date$/,
      },
    },
  },
  decorators: [
    (Story) => (
      <div
        className={`${inter.variable} ${openSans.variable} font-sans`}
      >
        {/* 👇 Decorators in Storybook also accept a function. Replace <Story/> with Story() to enable it  */}
        <Story />
      </div>
    ),
  ],
};

export default preview;

```

**Reply by JurajKavka** [reply]
I think, You don't need anything from Next project in storybook at all. Just use fonts directly in `preview.ts` with their imports, like you are doing it in `'./../src/pages/_app';` 

But I think that you have main next project and storybook installed over it in same repo with same package.json and dependencies?

In my case, I have standalone UI library in dedicated repository and NextJS is installed there, just because it needs to stand on "Next". So I don't have any code in `pages` or `app` directory. Everything is configured on storybook level in `preview.tsx`. I'm installig that UI library to he main "NextJS" project from private npm repo in out gitlab.

So You are doing it well, I just described how I work.

**Reply by kamal-choudhary** [reply]
Yes, my main NextJS project and Storybook are in the same repo.

**Comment by colmukrainec** [comment]
Did you ever find a fix to this problem? I believe I am having this problem as well.

**Reply by jonathanasquier** [reply]
This works => https://towardsdev.com/configure-next-js-fonts-with-tailwind-in-storybook-f51541ad77de
Just put the decorator in a tsx file elsewhere
